mods/d2k/maps/ordos-06a/ordos06a.lua

```diff
@@ -0,0 +1,255 @@
+	normal = DateTime.Minutes(2) + DateTime.Seconds(15),
+	hard = DateTime.Minutes(1) + DateTime.Seconds(15)
+}
+
+AtreidesPaths =
+{
+	{ AtreidesEntry2.Location, AtreidesRally2.Location },
+	{ AtreidesEntry3.Location, AtreidesRally3.Location },
+	{ AtreidesEntry4.Location, AtreidesRally4.Location }
+}
+
+HarkonnenPaths =
+{
+	{ HarkonnenEntry2.Location, HarkonnenRally2.Location },
+	{ HarkonnenEntry3.Location, HarkonnenRally3.Location },
+	{ HarkonnenEntry4.Location, HarkonnenRally4.Location },
+	{ HarkonnenEntry5.Location, HarkonnenRally5.Location },
+	{ HarkonnenEntry6.Location, HarkonnenRally6.Location },
+	{ HarkonnenEntry7.Location, HarkonnenRally7.Location }
+}
+
+AtreidesAttackWaves =
+{
+	easy = 3,
+	normal = 4,
+	hard = 5
+}
+
+HarkonnenAttackWaves =
+{
+	easy = 4,
+	normal = 5,
+	hard = 6
+}
+
+InitialReinforcements = 
+{
+	Atreides = { "combat_tank_a", "quad", "quad", "trike", "trike" },
+	Harkonnen = { "trooper", "trooper", "trooper", "trooper", "trooper", "combat_tank_h" }
+}
+
+InitialReinforcementsPaths =
+{
+	Atreides = { AtreidesEntry1.Location, AtreidesRally1.Location },
+	Harkonnen = { HarkonnenEntry1.Location, HarkonnenRally1.Location }
+}
+
+InitialContrabandTimes =
+{
+	easy = DateTime.Minutes(20),
+	normal = DateTime.Minutes(30),
+	hard = DateTime.Minutes(40)
+}
+
+ContrabandTimes =
+{
+	easy = DateTime.Minutes(5),
+	normal = DateTime.Minutes(6),
+	hard = DateTime.Minutes(7)
+}
+
+SendContraband = function()
+	UserInterface.SetMissionText("Ixian reinforcements have arrived!", player.Color)
+	Media.PlaySpeechNotification(player, "Reinforce")
+
+	local entryLocation = CPos.New(82, playerStarports[1].Location.Y + 1)
+	local exitLocation = CPos.New(2, playerStarports[1].Location.Y + 1)
+	Reinforcements.ReinforceWithTransport(player, "frigate", IxianReinforcements[Difficulty], { entryLocation, playerStarports[1].Location + CVec.New(1, 1) }, { exitLocation })
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		UserInterface.SetMissionText("")
+	end)
+
+	Trigger.AfterDelay(ContrabandTimes[Difficulty], function()
+		SendContraband()
+	end)
+end
+
+Hunt = function(house)
+	Trigger.OnAllKilledOrCaptured(Base[house.InternalName], function()
+		Utils.Do(house.GetGroundAttackers(), IdleHunt)
+	end)
+end
+
+CheckHarvester = function(house)
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[house] then
+		local units = house.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[house] = false
+			ProtectHarvester(units[1], house, AttackGroupSize[Difficulty])
+		end
+	end
+end
+
+Tick = function()
+	playerStarports = player.GetActorsByType("starport")
+
+	if #playerStarports < 1 and not player.IsObjectiveFailed(GuardStarport) then
+		player.MarkFailedObjective(GuardStarport)
+		atreides.MarkCompletedObjective(KillStarport1)
+		harkonnen.MarkCompletedObjective(KillStarport2)
+	end
+
+	if atreides.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillHarkonnen) then
```

Comment:
In this line `KillHarkonnen` should be changed to `KillAtreides` else it is not possible to finish the mission if the player defeats the Harkonnen first.
```diff
-       if atreides.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillHarkonnen) then
+       if atreides.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillAtreides) then
                Media.DisplayMessage("The Atreides have been annihilated!", "Mentat")
                player.MarkCompletedObjective(KillAtreides)
        end
```

Reply:
That should also fix the looping message.